src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -366,9 +366,10 @@ export class ActivityGraphFillComputer {
     let pixelEnd = (sampleEnd - rangeStart) * xPixelsPerMs;
     pixelStart = Math.max(0, pixelStart);
     pixelEnd = Math.min(canvasPixelWidth - 1, pixelEnd);
+    const pixelCenter = (sampleTime - rangeStart) * xPixelsPerMs;
     const intPixelStart = Math.floor(pixelStart);
     const intPixelEnd = Math.floor(pixelEnd);
-    const intPixelCenter = Math.floor((intPixelStart + intPixelEnd) / 2);
+    const intPixelCenter = Math.floor(pixelCenter);
```

Comment:
I think it would make sense to rename some variables here.

 - `sampleStart` -> `sampleCategoryStartTime`
 - `sampleEnd` -> `sampleCategoryEndTime`
 - `pixelStart` -> `sampleCategoryStartPixel`
 - `pixelEnd` -> `sampleCategoryEndPixel`
 - `pixelCenter` -> `samplePixel`
 - `intPixelStart` -> `intCategoryStartPixel`
 - `intPixelEnd` -> `intCategoryEndPixel`
 - `intPixelCenter` -> `intSamplePixel`
 - `sampleFirstHalfRatio` -> `beforeSampleCpuRatio`
 - `sampleSecondHalfRatio` -> `afterSampleCpuRatio`

Also, I believe the code in the rest of this function is not correct. It's trying to accumulate the contributions in one big range, with the "partial pixels" subtracted at the very end. I believe it'll be more accurate to treat the "beforeSample" and "afterSample" intervals completely separately, and accumulate two ranges (with two "partial pixel" subtractions per range (four in total)). But we can fix that later.

Reply:
The most important aspect of these renames, to me, is to make it clear that the sample's category contribution interval can be off-balance; the sample is not necessarily in the center and the two parts aren't necessarily halves.

Reply:
Thanks for the renaming recommendations. These are better indeed. I just renamed them. You are also right about the subtraction. I just filed #3425 for that, and will send a PR for that tomorrow.